src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -560,15 +583,33 @@ class MessageHelperTest : public ::testing::Test {
       , delta_from_functions_id(32768) {}
 
  protected:
-  application_manager_test::MockApplicationManager mock_application_manager;
+  MockApplicationManager mock_application_manager;
   const StringArray language_strings;
   const StringArray hmi_result_strings;
   const StringArray mobile_result_strings;
   const StringArray function_id_strings;
   const StringArray events_id_strings;
   const StringArray hmi_level_strings;
-
   const size_t delta_from_functions_id;
+  void CheckMessageParams(mobile_apis::SoftButtonType::eType button_type) {
+    smart_objects::SmartObject soft_button;
+    soft_button[strings::system_action] =
+        mobile_apis::SystemAction::INVALID_ENUM;
+    soft_button[strings::type] = button_type;
+    smart_objects::SmartObject message_params;
+    message_params[strings::soft_buttons][0] = soft_button;
+
+    MockApplicationSharedPtr app_shared_ptr = utils::MakeShared<AppMock>();
+    policy_test::MockPolicyHandlerInterface policy_handler;
+
+    EXPECT_CALL(policy_handler, PolicyEnabled()).WillOnce(Return(true));
+    EXPECT_CALL(policy_handler, CheckSystemAction(_, _)).WillOnce(Return(true));
```

Comment:
@indlu Are the arguments of the call to `CheckSystemAction()` irrelevant? `soft_button[strings::system_action]` is set.